pandas/core/aggregation.py

```diff
@@ -5,14 +5,84 @@
 
 from collections import defaultdict
 from functools import partial
-from typing import Any, DefaultDict, List, Sequence, Tuple
+from typing import (
+    Any,
+    Callable,
+    DefaultDict,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+import numpy as np
 
 from pandas.core.dtypes.common import is_dict_like, is_list_like
+from pandas.core.dtypes.generic import ABCDataFrame, ABCSeries
 
+from pandas.core.base import SpecificationError
 import pandas.core.common as com
 from pandas.core.indexes.api import Index
 
 
+def reconstruct_func(
+    func: Optional[Union[List, Dict]], **kwargs
```

Comment:
Why is func an optional List / Dict? Should be a callable, no? If so can you add subtypes to clarify?

Reply:
Can you clarify this? I am pretty confused reading the annotation versus the docstring and implementation as to what this is

Reply:
Thanks very much for the question, sorry to miss this one!

And indeed, this is very confusing @WillAyd 

The possible scenarios are:
1. normal aggregations, then `func` could be either str, or Callable, or list of them, or the dictionary of both
2. named aggregation, then `func` will be `None`, and all those information are parsed to `kwargs`

I added typings for `func`, and also add more descriptive comments in the docstring to help understand.

Very sorry for this unclear docstring and functionality. And all your further feedbacks or comments are very welcome and appreciated @WillAyd